sdk/internal/shared/shared.go

```diff
@@ -0,0 +1,34 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package shared
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/errorinfo"
+)
+
+// NonRetriableError marks the specified error as non-retriable.
+func NonRetriableError(err error) error {
```

Comment:
Might simplify things to move this to the `errorinfo` package so this and `NonRetriable` are together.